pkg/scheduler/reconciliation/reconciliation_suite_test.go

```diff
@@ -0,0 +1,125 @@
+package reconciliation
+
+import (
+	"context"
+	"github.com/kyma-incubator/reconciler/pkg/cluster"
+	"github.com/kyma-incubator/reconciler/pkg/db"
+	"github.com/kyma-incubator/reconciler/pkg/keb/test"
+	"github.com/kyma-incubator/reconciler/pkg/logger"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"path/filepath"
+	"sync"
+	"testing"
+)
+
+type reconciliationTestSuite struct {
```

Comment:
if you split the transaction relevant tests (e.g. rollback checks) into a second test suite, all others could run with the inbuilt Tx Connection I think, but I will double check this on a second review. Not sure if we want this at all, just a thought as you implemented this with an isolated suite and custom Connection methods

Reply:
I thought about it too, IMO it would an overkill, since most tests require multiple transactions (aka dbConnection) for tests. But sure we could think about it an a follow-up PR I'd say.